rest-services/server/src/main/java/com/_4point/aem/docservices/rest_services/server/assembler/AssembleDocuments.java

```diff
@@ -0,0 +1,209 @@
+			try (Document result = executeOn(ddx, mapOfDocuments, assemblerService)) {
+				String contentType = result.getContentType();
+				ServletUtils.validateAcceptHeader(request.getHeader(AcceptHeaders.ACCEPT_HEADER_STR), contentType);
+				response.setContentType(contentType);
+				response.setContentLength((int) result.length());
+				ServletUtils.transfer(result.getInputStream(), response.getOutputStream());
+			}
+		} catch (AssemblerServiceException | IOException e) {
+			// TODO Auto-generated catch block
+			throw new InternalServerErrorException("Internal Error while merging PDF. (" + e.getMessage() + ").", e);
+		} catch (Exception e) {
+			log.error("Error while Merging pdfs " + e.getMessage());
+		}
+
+	}
+
+	private Document executeOn(Document ddx, Map<String, Object> mapOfDocuments, AssemblerService assemblerService)
+			throws AssemblerServiceException, IOException {
+		// TODO Auto-generated method stub
+        AssemblerOptionsSpec assemblerOptionSpec = new AssemblerOptionsSpecImpl();
+		log.info("FailonError=" + assemblerOptionSpec.isFailOnError());
+
+		AssemblerResult assemblerResult = assemblerService.invoke(ddx, mapOfDocuments, assemblerOptionSpec);
+		
+		Map<String, Document> allDocsReturned = assemblerResult.getDocuments();
+
+		Document concatenatedDoc = null;
+		// byte[] concatenatedPDF = null;
+
+		for (Entry<String, Document> docsMap : allDocsReturned.entrySet()) {
+			String concatenatedPDFName = (String) docsMap.getKey();
+			if (concatenatedPDFName.equalsIgnoreCase("concatenatedPDF.pdf")) {
+				Object pdf = docsMap.getValue();
+				concatenatedDoc = (Document) pdf;
+				// concatenatedPDF =
+				// org.apache.commons.io.IOUtils.toByteArray(concatenatedDoc.getInputStream());
+				break;
+			}
+		}
+		return concatenatedDoc;
+	}
+
+	private InputStream createDdxFile(SlingHttpServletRequest request, Map<String, Object> mapOfDocuments)
+			throws Exception {
+		InputStream xmlInputStream = null;
+
+		final boolean isMultipart = org.apache.commons.fileupload.servlet.ServletFileUpload.isMultipartContent(request);
+		DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder docBuilder = null;
+		try {
+			docBuilder = docBuilderFactory.newDocumentBuilder();
+			org.w3c.dom.Document ddx = docBuilder.newDocument();
+			Element mainRootElement = ddx.createElementNS("http://ns.adobe.com/DDX/1.0/", "DDX");
+			ddx.appendChild(mainRootElement);
+			mainRootElement.appendChild(getPDFNodes(ddx, isMultipart, mapOfDocuments, request));
+
+			DOMSource domSource = new DOMSource(ddx);
+			ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+			Result outputTarget = new StreamResult(outputStream);
+
+			TransformerFactory.newInstance().newTransformer().transform(domSource, outputTarget);
+			xmlInputStream = new ByteArrayInputStream(outputStream.toByteArray());
+
+		} catch (Exception e) {
+			throw new Exception("Error creating ddx xml file", e);
```

Comment:
While catching Exception is usually wrong (but not always) as noted in my earlier comment, throwing Exception is *always* wrong.  You should always throw some subclass of Exception, not an Exception object itself.  Here's an article that covers [common mistakes when handling exceptions](https://stackify.com/common-mistakes-handling-java-exception/).